ext/dom/document.c

```diff
@@ -1734,8 +1735,8 @@ static void _dom_document_relaxNG_validate(INTERNAL_FUNCTION_PARAMETERS, int typ
 	switch (type) {
 	case DOM_LOAD_FILE:
 		if (CHECK_NULL_PATH(source, source_len)) {
-			php_error_docref(NULL, E_WARNING, "Invalid RelaxNG file source");
-			RETURN_FALSE;
+			zend_argument_value_error(1, "must not contain any null bytes");
```

Comment:
It looks like the source_len == 0 case above this one is still left.

Reply:
Ah I missed that one, thanks. While giving it a second look, I also noticed other invalid path checks that should throw IMHO. I added these in a separate commit.

Also, I added some unit tests to improve coverage.